### ISSUE TITLE
Rule proposal: `no-array-for-each`, add option to ignore for optional chaining

### ISSUE THREAD

**Issue by aedryan** [sindresorhus/eslint-plugin-unicorn#1744]
### Description

`no-array-for-each` is neat but I think readability could be improved by allowing a fall back to `.forEach` when optional chaining is desired.  I personally think that this is more readable:
```
arr?.forEach(() => {
  // do something
});
```
Than this:
```
if (arr) {
  for (const a of arr) {
    // do something
  }
}
```
It would be nice for myself and folks that agree to have an option for `no-array-for-each` to pass code that uses `.forEach` with optional chaining.

### Fail

This currently fails.
```js
arr?.forEach(() => {
  // do something
});
```

### Pass

I believe the same failure above should pass with something like a `permitOptionalChaining: true` option.
```js
arr?.forEach(() => {
  // do something
});
```

**Comment by sindresorhus** [comment]
You can just do:

```js
for (const element of foo?.bar ?? []) {

}
```

**Comment by aedryan** [comment]
Indeed, in my opinion, I think guarding to empty iterables like that looks less clean, like it's covering for a mistake or an unknown.
